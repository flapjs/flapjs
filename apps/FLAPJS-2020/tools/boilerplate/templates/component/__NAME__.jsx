import React from 'react';
import PropTypes from 'prop-types';
// import Style from './__NAME__.module.css';

/**
 * A React component that can do anything you want. :D
 */
class __NAME__ extends React.Component
{
    constructor(props)
    {
        super(props);
    }

    /** @override */
    render()
    {
        const props = this.props;
        
        return (
            <div id={props.id}
                style={props.style}
                className={props.className}>
            </div>
        );
    }
}

__NAME__.propTypes = {
    id: PropTypes.string,
    style: PropTypes.object,
    className: PropTypes.string,
};
__NAME__.defaultProps = {
};

export default __NAME__;
