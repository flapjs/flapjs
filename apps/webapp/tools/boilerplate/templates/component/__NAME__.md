# __NAME__

## Motivation
Cause I love all of you. <3

## Work Log
- Created __NAME__ component.