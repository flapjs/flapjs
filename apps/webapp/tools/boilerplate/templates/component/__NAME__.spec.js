import React from 'react';
import { shallow } from 'enzyme';

import __NAME__ from './__NAME__.jsx';

let wrapper;

beforeEach(() =>
{
    wrapper = shallow(<__NAME__/>);
});

describe('<__NAME__/> rendering', () =>
{
    test('renders correctly', () =>
    {
        expect(wrapper).toMatchSnapshot();
    });
});

describe('<__NAME__/> interaction', () =>
{
    /** I promise I will test you soon... */
});

describe('<__NAME__/> lifecycle', () =>
{
    /** ...later. */
});
