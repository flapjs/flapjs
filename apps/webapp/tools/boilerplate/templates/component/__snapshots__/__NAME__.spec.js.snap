// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<__NAME__/> rendering renders correctly 1`] = `undefined`;
